api/api-agents-v4/src/main/java/com/thoughtworks/go/apiv4/agents/AgentsControllerV4.java

```diff
@@ -83,134 +51,16 @@ public void setupRoutes() {
         path(controllerBasePath(), () -> {
             before("", mimeType, this::setContentType);
             before("/*", mimeType, this::setContentType);
-            before("", mimeType, this::checkSecurityOr403);
-            before("/*", mimeType, this::checkSecurityOr403);
 
-            get("", mimeType, this::index);
-            get(Routes.AgentsAPI.UUID, mimeType, this::show);
-            patch(Routes.AgentsAPI.UUID, mimeType, this::update);
-            patch("", mimeType, this::bulkUpdate);
-            delete(Routes.AgentsAPI.UUID, mimeType, this::deleteAgent);
-            delete("", mimeType, this::bulkDeleteAgents);
+            before("", mimeType, apiAuthenticationHelper::checkAdminUserAnd403);
+            before("/*", mimeType, apiAuthenticationHelper::checkAdminUserAnd403);
 
-            exception(HttpException.class, this::httpException);
+            before("", mimeType, (req, resp) -> {
+                throw haltBecauseOfUnsupportedAPIVersion(mimeType, UNSUPPORTED_API_NAME);
```

Comment:
This API will return `415` instead of `404`. 
Another question, will this controller stay forever?

Reply:
Hi @GaneshSPatil, there is a slack discussion thread around exactly this topic which I just now shared with you. That discussion was not conclusive though.

I am ok with whatever we conclude to go ahead with as every company is handling this kind of situation differently. Google uses `400 (bad request)` and twitter is using `410 (Gone)` status code for such cases.

@maheshp, @ketan, @arvindsv - Any thoughts?

Reply:
Hmm. My worry is with the older APIs which are removed.
For Agents API,
`v1`, `v2`, `v3` will say `404`(Not Found), whereas, `v4` will say `415`, which seemed a little odd to me.

Reply:
I am ok keeping the existing behaviour and return a 404.

Reply:
@msaurabh1978 @GaneshSPatil @maheshp @ketan @arvindsv 
This discussion is not a direct part of this story and will be decided for all the deprecated APIs separately. For now, following the old pattern - we have removed the code for Envs v2 and Agents v4 and v5 - which will render `404`.